src/routing.cpp

```diff
@@ -125,6 +126,22 @@ List routing(DateVector SimPeriod, NumericMatrix surfaceRunoff, NumericMatrix Gr
 	numbers =  findUniqueValues(routeOrder); //finding unique values in route order (ascending) => routingSteps
 	numbersSorted = sortIt(numbers);
 
+	// If K_release is necessary (ie we use Hanasaki algrithm) then initialize it
+	if (ReservoirType != 1) 
+	{
+		for (int cell = 0; cell < array_size; cell++)
+		{
+			if (S_ResStorage[cell] < G_STORAGE_CAPACITY[cell] * 1000. * 1000. * 0.1)
+			{
+				K_release[cell] = 0.1;
+			}
+			else
+			{
+				K_release[cell] = S_ResStorage[cell] / (G_STORAGE_CAPACITY[cell] * 1000. * 1000.);
+			}
+		}
+
+	}
```

Comment:
Actually, I would prefer using the simple .fill(0.1) option as it is faster (so you do not need the if condition), and your option also has the drawback that the release is defined when the operational year has not yet started. But we can stick with your solution. I do not think that the simulation is highly influenced because it should only affect the first months of the simulation period until the Hanasaki function overwrites it again. 

A small side note, what I've just noticed: Isn't it that the K_relase definition can be re-written in the form of: 
double MIN_RELEASE =  0.1f;
double KM3_to_MMKM2 = 1000. * 1000.;
K_release[cell] = max(S_ResStorage[cell] / (G_STORAGE_CAPACITY[cell] * KM3_to_MMKM2 ), MIN_RELEASE );